### DISCUSSION TITLE
David Mytton - ArcJet

### DISCUSSION THREAD

**Discussion by hipstersmoothie** [devtools-fm/devtools.fm#337] [Announcements]
- https://davidmytton.blog/
- Co-founder of https://console.dev/
- Investing
- https://www.linkedin.com/in/davidmytton/
- https://x.com/davidmytton
- https://arcjet.com/
- https://github.com/davidmytton

## Questions

- [ ] ~You've invested in a lot of devtools companies. What do you look for when making investments?~

### Enviroment

- [x] What is sustainable computing?
- [x] What do you think are some of the most wasteful sectors of computing?
- [x] This seems like a problem that could only balloon. How do you think we might get out of this mess?

### Arcjet

- [x] What is arcjet?
- [x] What are some of the different things that arcjet helps protect you from with Shield?
- [x] How is it different from than established solutions like a Web Application Firewall?
- [x] Doing rate limiting / bot detection in process has some drawbacks (e.g. the requests still hit your service). Are there future plans to provide infrastructure that people can put in front of their application?
- [x] It seems like some of arcjet is behind an API. What is the API doing that the local client can't?
- [x] The docs mention WASM as a way arcjet checks requests. What are you using WASM for? Why choose it?
- [x] How do you keep my app fast when I have to call your api? What happens to my app if you have an outage?
- [x] You've built in bot protection. What were some challenges in building that out? How do you actually detect something is a bot?
- [ ] ~Is it a panacea for security problems? Or does it still have some limitiations?~
- [x] What do you think the software industry is getting wrong about security?

### Future

- [x] What next for arcjet? what does 1.0 look like?
- [ ] What do you think the future of security in software applications looks like?

**Comment by just-be-dev** [comment]
Competitors of note: Data dome (NYT uses them for bot detection), CF
